enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/CommonOtpService.java

```diff
@@ -135,6 +145,17 @@ public OtpVerifyResponse verifyOtpCode(String processId, OwnerId ownerId, String
         return response;
     }
 
+    private int fetchFailedAttemptsCount(final OnboardingProcessEntity process, final OtpType otpType) throws OnboardingProcessException {
+        if (otpType == OtpType.ACTIVATION) {
+            return onboardingOtpRepository.countFailedAttempts(process, otpType);
+        } else {
+            final String activationId = process.getActivationId();
+            final IdentityVerificationEntity identityVerification = identityVerificationRepository.findFirstByActivationIdOrderByTimestampCreatedDesc(activationId)
+                    .orElseThrow(() -> new OnboardingProcessException("Identity verification not found, activation ID: " + activationId));
+            return onboardingOtpRepository.countFailedAttempts(process, otpType, identityVerification);
```

Comment:
ℹ️ This is the main change of this PR. OTP failed attempts are count per the whole proces for ACTIVATION, and per process and identity verification (because of possible restart) for USER_VERIFICATION.

Reply:
Good.